leasing/models/invoice.py

```diff
@@ -18,3 +18,14 @@ class Invoice(TimestampedModelMixin):
                                         blank=True, on_delete=models.PROTECT)
     state = EnumField(InvoiceState, verbose_name=_("State"), max_length=255,
                       default=InvoiceState.PENDING)
+
+    def create_reference_number(self):
+        if not self.id:
+            return None
+
+        reference_number = '91112{}880'.format(self.id)
+
+        self.reference_number = reference_number + str(
+            (10 - (sum((7, 3, 1)[i % 3] * x for i, x in enumerate(map(int, str(reference_number)[::-1]))) % 10)) % 10)
```

Comment:
It's a nice one liner, but I would say that couple temp variables would make this a bit easier to read though. :)

```python
reversed_digits = reversed(str(reference_number))
checksum = -sum((7, 3, 1)[i % 3] * int(x) for (i, x) in enumerate(digits)) % 10
self.reference_number = reference_number + str(checksum)